src/macros/src/lib.rs

```diff
@@ -0,0 +1,18 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn;
+
+#[proc_macro_derive(DefaultDoCommand)]
+pub fn trivially_generic_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
+    impl_do_command(&ast)
+}
+
+fn impl_do_command(ast: &syn::DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+    let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
+    let gen = quote! {
+        impl #impl_generics DoCommand for #name #ty_generics #where_clause {}
```

Comment:
Does `DoCommand` here need to be qualified somehow so that it is clear that it refers to the micro-rdk `DoCommand`? What would happen if I used this macro in a module?

Reply:
you have to import the DoCommand trait to use it, but I suppose I could replace `DoCommand` with `micro_rdk::common::generic::DoCommand`. Is that preferred?

Reply:
I'm actually not sure if that is better or not. Because, at least in theory, the micro-rdk could actually be imported under a different name, with something like:
```
[dependencies]
micro-rdk = { git = "https://github.com/viamrobotics/micro-rdk.git", features = ["esp32"], package = "petit" }
```

 So maybe we don't actually know what the right top-level name is. Is there a way to use a relative name: `<magic>::common::generic::DoCommand` that indicates that we always want `common::generic::DoCommand` from our containing crate? `super`?

Reply:
Yeah I think it is fine to require that `DoCommand` be imported. Not ideal, but fine. Resolving.